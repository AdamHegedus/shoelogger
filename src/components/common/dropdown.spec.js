import { shallowMount, createLocalVue } from '@vue/test-utils';
import BootstrapVue from 'bootstrap-vue';
import Dropdown from '@/components/common/dropdown';

const localVue = createLocalVue();
localVue.use(BootstrapVue);

describe('dropdown.vue', () => {
    describe('render', () => {
        it('should render component when selected is null', () => {
            // ASSIGN
            const wrapper = shallowMount(Dropdown, {
                localVue,
                propsData: {
                    event: 'mock-event',
                    placeholder: 'mock-placeholder',
                    selected: null,
                    data: [
                        {
                            id: 0,
                            text: 'Mock 01'
                        },
                        {
                            id: 1,
                            text: 'Mock 02'
                        }
                    ],
                    getDisplayValue: (item) => { return item.text; }
                }
            });

            // ACT
            const actual = wrapper.element;

            // ASSERT
            expect(actual).toMatchSnapshot();
        });

        it('should render component when selected is set', () => {
            // ASSIGN
            const wrapper = shallowMount(Dropdown, {
                localVue,
                propsData: {
                    event: 'mock-event',
                    placeholder: 'mock-placeholder',
                    selected: {
                        id: 0,
                        text: 'Mock 01'
                    },
                    data: [
                        {
                            id: 0,
                            text: 'Mock 01'
                        },
                        {
                            id: 1,
                            text: 'Mock 02'
                        }
                    ],
                    getDisplayValue: (item) => {
                        return item.text;
                    }
                }
            });

            // ACT
            const actual = wrapper.element;

            // ASSERT
            expect(actual).toMatchSnapshot();
        });
    });

    describe('select', () => {
        it('should emit event', () => {
            // ASSIGN
            const wrapper = shallowMount(Dropdown, {
                localVue,
                propsData: {
                    event: 'mock-event',
                    placeholder: 'mock-placeholder',
                    selected: null,
                    data: [
                        {
                            id: 0,
                            text: 'Mock 01'
                        },
                        {
                            id: 1,
                            text: 'Mock 02'
                        }
                    ],
                    getDisplayValue: (item) => {
                        return item.text;
                    }
                }
            });
            const input = {
                id: 0,
                text: 'Mock 01'
            };
            const mockEvent = jest.fn();
            wrapper.vm.$on('mock-event', mockEvent);

            // ACT
            wrapper.vm.select(input);

            // ASSERT
            expect(mockEvent).toBeCalledWith(input);
        });
    });
});
