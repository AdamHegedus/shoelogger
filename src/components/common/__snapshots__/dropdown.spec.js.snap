// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dropdown.vue render should render component when selected is null 1`] = `
<b-dropdown-stub
  boundary="scrollParent"
  class="btn-block"
  id="mock-event"
  no-flip="true"
  role="menu"
  size="lg"
  text="mock-placeholder"
  toggleclass="btn btn-secondary btn-lg btn-block shadow"
  toggletext="Toggle Dropdown"
>
  <b-dropdown-item-stub
    activeclass="active"
    class="[object Object]"
    event="click"
    exactactiveclass="active"
    routertag="a"
    target="_self"
  >
    
        Mock 01
    
  </b-dropdown-item-stub>
  <b-dropdown-item-stub
    activeclass="active"
    class="[object Object]"
    event="click"
    exactactiveclass="active"
    routertag="a"
    target="_self"
  >
    
        Mock 02
    
  </b-dropdown-item-stub>
</b-dropdown-stub>
`;

exports[`dropdown.vue render should render component when selected is set 1`] = `
<b-dropdown-stub
  boundary="scrollParent"
  class="btn-block"
  id="mock-event"
  no-flip="true"
  role="menu"
  size="lg"
  text="Mock 01"
  toggleclass="btn btn-secondary btn-lg btn-block shadow"
  toggletext="Toggle Dropdown"
>
  <b-dropdown-item-stub
    activeclass="active"
    class="[object Object]"
    event="click"
    exactactiveclass="active"
    routertag="a"
    target="_self"
  >
    
        Mock 01
    
  </b-dropdown-item-stub>
  <b-dropdown-item-stub
    activeclass="active"
    class="[object Object]"
    event="click"
    exactactiveclass="active"
    routertag="a"
    target="_self"
  >
    
        Mock 02
    
  </b-dropdown-item-stub>
</b-dropdown-stub>
`;
